lib/adapter.js

```diff
@@ -3236,7 +3240,7 @@ function Adapter(options) {
                             _cnt++;
                             adapterObjects.getObjectView('system', 'state', {startkey: deviceName + '.', endkey: deviceName + '.\u9999'}, options, (err, res) => {
                                 _cnt--;
-                                if (err) {
+                                if (err || !res || !res.rows) {
                                     typeof callback === 'function' && callback(err);
                                     callback = null;
```

Comment:
I know you didn't touch that code. But what is this supposed to do?

Reply:
I found this construct in several places where the callback is only having the error parameter. In this case delete* methods. And the semantics would be: if an "err" was set it is passed  through. if no error is set but also no result data exists the callback in fact is called with an falsy err variable which is the normal success case in this case.

Not nice but should be ok

Reply:
No I mean `callback = null;`

Reply:
this is to make sure that also in the "loop over stuff" the callback is only called once... or what is the questions. But yes on top unneeded because we return directly. done